test/fast/Unit tests/nvm_supports_xz

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+cleanup () {
+  rm -rf ../../xz-test
+}
+die () { echo $@ ; exit 1; }
+
+. ../../../nvm.sh
+
+mkdir ../../xz-test
+touch ../../xz-test/xz
+chmod +x ../../xz-test/xz
+ln -s /usr/bin/which ../../xz-test/which
+ln -s /usr/bin/command ../../xz-test/command
+ln -s /usr/bin/awk ../../xz-test/awk
+ln -s $(which rm) ../../xz-test/rm
+
+
+OLDPATH=$PATH
+TEST_PATH=../../xz-test
+export PATH=$TEST_PATH:$PATH
+
+$(nvm_supports_xz "v4.0.0")
+EXIT_CODE_SUPPORTED=$?
+
+$(nvm_supports_xz "v0.12.7")
+EXIT_CODE_TOO_OLD=$?
+
+
+# Emulate a missing xz binary
+rm ../../xz-test/xz
+export PATH=$TEST_PATH
+$(nvm_supports_xz "v4.0.0")
+EXIT_CODE_MISSING_XZ=$?
+
+export PATH=$OLDPATH
+cleanup
+
+[ "_$EXIT_CODE_SUPPORTED" = "_0" ] || \
+  die "expected 'nvm_supports_xz v4.0.0' to exit with 0, got $EXIT_CODE"
```

Comment:
That is true - and that's why `cleanup` is a function :-) however, if you use a `teardown` file, that will be called even when the test bails early, so that might be the most robust approach?